### PR TITLE
Wrap RCT2's addresses in RCT2_ADDRESS macro

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/management/news_item.c [hunk 1]
```diff
@@ -278,7 +278,7 @@ void news_item_add_to_queue_raw(uint8 type, const utf8 *text, uint32 assoc)
 
 	// find first open slot
 	while (newsItem->type != NEWS_ITEM_NULL) {
-		if (newsItem + 1 >= (rct_news_item*)0x13CB1CC) // &news_list[10]
+		if (newsItem + 1 >= &gNewsItems[10]) // &news_list[10]
```
Comment (janisozaur) [inline src/management/news_item.c]: This is the only non-`RCT2_ADDRESS`, non-`RCT2_GLOBAL` item in this PR

File: src/windows/ride.c [hunk 2]
```diff
@@ -2507,8 +2507,8 @@ static rct_string_id window_ride_get_status_overall_view(rct_window *w, void *ar
 	rct_string_id stringId;
 
 	ride_get_status(w->number, &formatSecondary, &argument);
-	RCT2_GLOBAL((uintptr_t)arguments + 0, uint16) = formatSecondary;
```
Comment (janisozaur) [inline src/windows/ride.c]: These pointers are not RCT2-specific, they get passed on from caller

**Comment by janisozaur** [comment]
This is ready now, it should cover all RCT2's addresses

File: src/management/news_item.c [hunk 3]
```diff
@@ -407,7 +407,7 @@ void news_item_add_to_queue_custom(rct_news_item *newNewsItem)
 
 	// Find first open slot
 	while (newsItem->type != NEWS_ITEM_NULL) {
-		if (newsItem + 1 >= (rct_news_item*)0x013CB1CC)
+		if (newsItem + 1 >= RCT2_ADDRESS(0x013CB1CC, rct_news_item))
```
Comment (marijnvdwerf) [inline src/management/news_item.c]: Isn't this the same as the change above?

**Pull request merged**